backend/curation/api/v1/curation/collections/common.py

```diff
@@ -259,6 +265,61 @@ def reshape_dataset_for_curation_api(
     return ds
 
 
+def reshape_dataset_for_curation_datasets_index_api(
+    visibility: str,
+    collection_version: Union[CollectionVersionWithPublishedDatasets, CollectionVersionWithPrivateDatasets],
+    dataset_version: DatasetVersion,
+) -> dict:
+    """
+    Create the response shape for the curation datasets index API response. Handles shape for both public and private
+    requests.
+    :param visibility: the requested visibility of the datasets to be included in the dataset index response; either PUBLIC or PRIVATE.
+    :param collection_version: the collection version of the dataset to be included in the API response.
+    :param dataset_version: a dataset version to be included in the API response.
+    :return: A dictionary shaped for inclusion in the datasets index API response.
+    """
+    # Determine visibility of dataset; used to calculate, and added to, dataset shape.
+    dataset_visibility = calculate_dataset_visibility(visibility, collection_version, dataset_version)
+
+    # Create base dataset response shape. use_canonical is true only for datasets with a calculated visibility of
+    # public; for datasets with a calculated visibility of private, corresponding fields are calculated below.
+    is_dataset_visibility_public = dataset_visibility == DatasetVisibility.PUBLIC.name
+    ds = reshape_dataset_for_curation_api(dataset_version, index=True, use_canonical_url=is_dataset_visibility_public)
```

Comment:
can we use `dataset` as the variable name here instead of `ds`? Or perhaps `dataset_response` or something like that? `ds` is vague.

Reply:
I agree `ds` is vague but I am following the naming convention in the existing [reshape_dataset_for_curation_api](https://github.com/chanzuckerberg/single-cell-data-portal/blob/main/backend/curation/api/v1/curation/collections/common.py#L209); are you OK with breaking consistency here?

Reply:
I'm hoping I didn't introduce that notation! 😏 

Do you want to change to `dataset` here? We can change the code other places in another refactor. If you don't want to refactor your code, you can dismiss this comment for now.

Reply:
Thanks Dan, I might leave it for now and we can address it together with the reference above in a refactor.